### PR TITLE
Remove Warnings from Ignored tests

### LINKED ISSUE #867: Remove Warnings from Ignored tests
Before releasing the RC, we should get rid of all Ignored tests, which show up as a warning in our build.
In some cases, this may mean making a feature work. In others it may mean revising the test.

### PR DESCRIPTION
**Pull request by rprouse**
~~**Do Not Merge**~~

Hold off merging until we decide if I should remove the culture tests and move other tests to the mock assembly.

Fixes #867 and #868 

For the ignored tests, 

- I deleted one that I originally wrote and can't think of a valid way to test.
- I comment out one test that is blocked by #352 and added a comment to that issue to that effect
- I switched an intermittent timing failure to explicit like the other timing issues so it won't cause a build warning

For the skipped tests,

- I reviewed all instances of the use of PlatformAttribute  or Include/Exclude.
- I removed a few that are no longer valid, or don't provide real tests
- See my comment on #868, I could delete the two Culture related tests since they are effectively only testing the behaviour of CultureInfo

This leaves 14 Explicit tests. These are divided into three categories, 3 long running tests, 7 that exercise failure conditions, I assume for development/debugging and the remaining 4 are intermittent failures.

I am thinking about moving the failing development/debugging tests to the mock assembly. That already has failing tests, so if people want to run them for debugging purposes, they can do so from there.

### CODE REVIEW

**Comment by rprouse** [comment]
I removed 3 explicit tests that aren't useful. The remaining 4 that are meant to fail cannot be move because they use TestBuilder. Should we comment them out in case someone needs to work with them in the future? If we did that, removed the 2 culture tests and maybe commented out the 3 EventPumpQueueLength tests, then the list of skipped tests would be down to 4 skipped tests that are intermittent failures. We could then add an issue to clean those up post 3.0.

**Comment by CharliePoole** [comment]
Your plan sounds good to me except for moving tests to the mock-assembly. That already had some tests moved into it that belong in TestData. Mock-assembly is for the explicit purpose of verifying how we handle test counts, failure counts, etc. and it has to be maintained with correct expected values. Everytime we change it we have to deal with re-calculating the expected numbers.

**Comment by rprouse** [comment]
I created #937 for the remaining 4 Explicit framework tests and fixed the console issue #936.

**Comment by rprouse** [comment]
This is ready to merge. 3 issues closer to RC :+1:

File: src/NUnitFramework/tests/Attributes/TheoryTests.cs [hunk 1]
```diff
@@ -96,13 +96,6 @@ public void EnumArgumentsAreSuppliedAutomatically()
...
         }
 
-        [Theory, Explicit("Used to demonstrate display of failing Theory")]
-        public void SquareRootWithAllBadValues(
-            [Values(-12.0, -4.0, -9.0)] double d)
-        {
-            SquareRootTest(d);
-        }
-
         [Datapoints]
```
Comment (CharliePoole) [inline src/NUnitFramework/tests/Attributes/TheoryTests.cs]: When I do the gui, I'll just create a separate assembly for stuff like this.

**Pull request merged**